### DISCUSSION TITLE
AutoFixture failure reproducibility question

### DISCUSSION THREAD

**Discussion by jzabroski** [AutoFixture/AutoFixture#1247] [Q&A]
I have some AutoFixture tests that generate object graphs that cause tests to fail sporadically, and it's next to impossible to reproduce (carefully reading the code to understand why it fails is often faster).  I was wondering if there was a way to log the seed data used by AutoFixture for a particular Fixture instance, so that it is easier to reproduce failures.

**Comment by aivascu** [comment]
@jzabroski there is no feature readily available for logging the seed I'm afraid. 
The current implementation of the random number generator does not use an explicit seed, under the hood it uses a `Random` instance initialized using the default constructor in its turn uses a time based seed. 
Also some generators (e.g. `StringGenerator`) use `Guid` to randomize the values.

Perhaps a better approach would be to log (serialize) the test data that's generated when the test fails?
